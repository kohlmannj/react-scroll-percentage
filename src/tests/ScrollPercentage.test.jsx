import React from 'react'
import { mount } from 'enzyme'
import ScrollPercentage from '../src/index.js'

jest.mock('invariant')

global.IntersectionObserver = jest.fn(() => ({
  observe: jest.fn(),
  unobserve: jest.fn(),
  disconnect: jest.fn(),
}))

beforeEach(() => {
  global.innerHeight = 800
})

afterEach(() => {
  global.innerHeight = 800
})

const plainChild = () => inner as div</div>

it('Should render <ScrollPercentage />', () => {
  const callback = jest.fn(plainChild)
  const wrapper = mount({callback} as ScrollPercentage</ScrollPercentage>)
  expect(callback).toHaveBeenLastCalledWith(
    expect.objectContaining({ inView: false }),
  )
  expect(wrapper).toMatchSnapshot()
})

it('Should render <ScrollPercentage /> with custom tag', () => {
  const callback = jest.fn(plainChild)
  const wrapper = mount(
    tag as ScrollPercentage="span" className="wrapperClass">
      {callback}
    </ScrollPercentage>,
  )
  expect(wrapper).toMatchSnapshot()
})

it('Should render with child', () => {
  const wrapper = mount(
    Inner as div as ScrollPercentage</div>
    </ScrollPercentage>,
  )
  expect(wrapper).toMatchSnapshot()
})

it('Should return a percentage', () => {
  expect(
    ScrollPercentage.calculatePercentage({ height: 250, bottom: 0 }),
  ).toEqual(1)
  expect(
    ScrollPercentage.calculatePercentage({ height: 250, bottom: 1050 }),
  ).toEqual(0)
})

it('Should handle threshold', () => {
  // 525 is half of 1050 - the full bottom size
  expect(
    ScrollPercentage.calculatePercentage({ height: 250, bottom: 525 }, 0.1),
  ).toEqual(0.5)
})
