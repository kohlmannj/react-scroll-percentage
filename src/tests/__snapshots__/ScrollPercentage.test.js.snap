// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render <ScrollPercentage /> 1`] = `
<ScrollPercentage
  tag="div"
  threshold={0}
>
  <Observer
    onChange={[Function]}
    threshold={0}
    triggerOnce={false}
  >
    <div>
      <div>
        inner
      </div>
    </div>
  </Observer>
</ScrollPercentage>
`;

exports[`Should render <ScrollPercentage /> with custom tag 1`] = `
<ScrollPercentage
  className="wrapperClass"
  tag="span"
  threshold={0}
>
  <Observer
    onChange={[Function]}
    threshold={0}
    triggerOnce={false}
  >
    <span
      className="wrapperClass"
    >
      <div>
        inner
      </div>
    </span>
  </Observer>
</ScrollPercentage>
`;

exports[`Should render with child 1`] = `
<ScrollPercentage
  tag="div"
  threshold={0}
>
  <Observer
    onChange={[Function]}
    threshold={0}
    triggerOnce={false}
  >
    <div>
      <div>
        Inner
      </div>
    </div>
  </Observer>
</ScrollPercentage>
`;
